documentation/head/replication.md

```diff
@@ -4,39 +4,37 @@ todo
 
 # Overview
 
-Postgres 9.4 (released in December 2014) introduced a new feature logical replication. Logical replication allow
-stream changes from database in real-time to external system.
+Postgres 9.4 (released in December 2014) introduced a new feature called logical replication. Logical replication allows
+changes from database to be streamed in real-time to external system. The difference between streaming replication and
+logical replication is that logical replication sends data over in a logical format whereas streaming replication sends data over in a binary format. Additionally logical replication can send over a single table, or database. Streaming replication was all or nothing.
 
-Previous for reach real-time update external system application can use "dual-writes".
-That is, every time your application code writes to the database,it also updates/invalidates the appropriate
-cache entries, reindexes the data in your search engine, sends it to your analytics system, and so on.
-Dual-writes approach is really problematic. It suffers from race conditions and reliability problems.
-If slightly different data gets written to two different datastores (perhaps due to a bug or a race condition),
-the contents of the datastores will gradually drift apart — they will become more and more inconsistent over time.
-Recovering from such gradual data corruption is difficult.
+Before Logical replication keeping an external system synchronized in real time was problematic. The application would have to  update/invalidate the appropriate cache entries, reindex the data in your search engine, send it to your analytics system, and so on.
```

Comment:
I want to say that without logical replication, application can use dual writes patterns - in that case application first write to database, then writes for example to cache syste, then to search engine etc and this pattern have many problems, because not all system supports transaction and for example after successfull writes to postgresql you can get error from elasticsearch that not support transactions, or successfully writes to elasticsearch but by some reason fail during writes to postgresql. In this case very difficult support consistence state in all system.